src/server/fiche-metier/infra/strapiFicheMetier.repository.ts

```diff
@@ -0,0 +1,25 @@
+import { StrapiRepository } from '~/server/cms/infra/repositories/strapi.repository';
+import { Strapi } from '~/server/cms/infra/repositories/strapi.response';
+import { Either } from '~/server/errors/either';
+import { FicheMétier } from '~/server/fiche-metier/domain/ficheMetier';
+import { FicheMetierRepository } from '~/server/fiche-metier/domain/ficheMetier.repository';
+import { mapFicheMetier } from '~/server/fiche-metier/infra/strapiFicheMetier.mapper';
+
+
+const RESOURCE_FICHE_METIER = 'fiche-metiers';
+export class StrapiFicheMetierRepository implements FicheMetierRepository{
+	constructor(private readonly strapiRepository: StrapiRepository) {
+	}
+	async getFicheMetierByNom(nom: string): Promise<Either<FicheMétier>> {
+		const query = `filters[nom_metier][$eq]=${encodeURIComponent(nom)}&populate=deep`;
+		const ficheMétierList = await this.strapiRepository.getCollectionType<Strapi.CollectionType.FicheMétier, FicheMétier>(RESOURCE_FICHE_METIER, query, mapFicheMetier);
+		return this.strapiRepository.getFirstFromCollection(ficheMétierList);
```

Comment:
🤔 C'est bizarre d'avoir ça dans le strapiRepository tel quel, ça n'a rien de spécifique au strapi. Je me demande si ça n'aurait pas plus sa place inliné ici ? Ou alors si c'est vraiement utilisé souvent avec strapi, peut-être merger avec l'appel juste au dessus : 
```suggestion
		const ficheMétierList = await this.strapiRepository.getFirstOf<Strapi.CollectionType.FicheMétier, FicheMétier>(RESOURCE_FICHE_METIER, query, mapFicheMetier);
```

Reply:
oui, une petite fonction getFirstOfCollection ça peut être bien comme c'est souvent utilisé !